cmd/osbuild-rcm-tests/main_test.go

```diff
@@ -41,40 +41,18 @@ func main() {
 	// Case 1: POST request
 
 	resp, err := http.Post(socket+endpoint, "application/json", strings.NewReader(submitBody))
-	if err != nil {
-		log.Fatal("Failed to submit a compose: ", err.Error())
-	}
-	if resp.StatusCode != 200 {
-		log.Print("Error: the ", endpoint, " returned non 200 status. Full response: ", resp)
-		failed = true
-	} else {
-		err = json.NewDecoder(resp.Body).Decode(&submitResponse)
-		if err != nil {
-			log.Fatal("Failed to decode JSON response from ", endpoint)
-		}
-		log.Print("Success: the ", endpoint, " returned compose UUID: ", submitResponse.UUID)
-	}
+	require.Nilf(t, err, "Failed to submit a compose: %v", err)
+	assert.Equalf(t, resp.StatusCode, 200, "Error: the %v returned non 200 status. Full response: %v", endpoint, resp)
+	err = json.NewDecoder(resp.Body).Decode(&submitResponse)
+	require.Nilf(t, err, "Failed to decode JSON response from %v", endpoint)
 
 	// Case 2: GET status
 
 	statusEndpoint := endpoint + "/" + submitResponse.UUID.String()
 	resp, err = http.Get(socket + statusEndpoint)
-	if err != nil {
-		log.Fatal("Failed to get a status: ", err.Error())
-	}
-	if resp.StatusCode != 200 {
-		log.Print("Error: the ", endpoint, " returned non 200 status. Full response: ", resp)
-		failed = true
-	} else {
-		err = json.NewDecoder(resp.Body).Decode(&statusResponse)
-		if err != nil {
-			log.Fatal("Failed to decode JSON response from ", endpoint)
-		}
-		log.Print("Success: the ", statusEndpoint, " returned status: ", statusResponse.Status)
-	}
+	require.Nilf(t, err, "Failed to get a status: %v", err)
+	assert.Equalf(t, resp.StatusCode, 200, "Error: the %v returned non 200 status. Full response: %v", endpoint, resp)
+	err = json.NewDecoder(resp.Body).Decode(&statusResponse)
+	require.Nilf(t, err, "Failed to decode JSON response from %v", endpoint)
 
-	// If anything failed return non-zero exit code.
-	if failed {
-		os.Exit(1)
-	}
```

Comment:
How do we report failure here? Does the testing framework return non-zero return code in case any `assert` failed?

Reply:
Yes, the idea is that returning non-0 from a test executable means it failed.

Reply:
Yes, I just verified that when an `assert` fails it exits with `1`

Reply:
ok, in that case it seems to be ready for merging as soon as the CI is happy

Reply:
> ok, in that case it seems to be ready for merging as soon as the CI is happy

Yeah for that we need to figure out an acceptable way to start the rcm socket.

Reply:
I think we should just assume someone does that for us (we do the same thing with `osbuild-composer` service currently) and we can think of a proper solution in a follow-up.

Reply:
I think @jrusz was referring to the CI failure below. Why is it even running in the CI? We don't yet have a support for running the tests in `/cmd`, do we?

Reply:
Oh, I understood that incorrectly. Integration tests are not meant to be run on Github Actions/Travis, so we just use a tag in these tests. When `go test ./...` is run, it ignores all the files which has a tag specified. See https://github.com/osbuild/osbuild-composer/blob/54e27e607ca865d1a7793640ca5f134daea13d45/cmd/osbuild-dnf-json-tests/main_test.go#L3

Just put `// +build integration` in `cmd/osbuild-rcm-tests/main_test.go` and everything should be alright.